server/controllers/entities/lib/resolver/resolve_external_ids.ts

```diff
@@ -1,49 +1,81 @@
-import { flatten, uniq } from 'lodash-es'
+import { flatten, uniqBy } from 'lodash-es'
 import { getInvEntitiesByClaim } from '#controllers/entities/lib/entities'
 import { getClaimValue } from '#controllers/entities/lib/inv_claims_utils'
-import { prefixifyWd } from '#controllers/entities/lib/prefix'
-import runWdQuery from '#data/wikidata/run_query'
-import { forceArray, objectEntries } from '#lib/utils/base'
-import type { Claims } from '#server/types/entity'
+import { prefixifyWd, prefixifyWdProperty } from '#controllers/entities/lib/prefix'
+import type { PropertyValuePair } from '#data/wikidata/queries/queries'
+import type { ResolvedExternalIdTriple } from '#data/wikidata/queries/resolve_external_ids'
+import { runWdQuery } from '#data/wikidata/run_query'
+import { forceArray, getOptionalValue, objectEntries } from '#lib/utils/base'
+import type { Claims, EntityUri, PropertyUri } from '#server/types/entity'
+import type { PropertyValueConstraints } from '#server/types/property'
 import { getInvEntityCanonicalUri } from '../get_inv_entity_canonical_uri.js'
 import { propertiesValuesConstraints as properties } from '../properties/properties_values_constraints.js'
 
-export async function resolveExternalIds (claims: Claims, resolveOnWikidata = true) {
-  const externalIds = []
+interface ResolveExternalIdsOptions {
+  resolveOnWikidata?: boolean
+  resolveLocally?: boolean
+  refresh?: boolean
+}
+
+interface FormattedResult {
```

Comment:
A more context specific proposition (already in use in `resolve_edition.ts#L13`:

```suggestion
interface ResultsFoundByExternalIds {
```

Reply:
In that case, we have a more specific name at hand 6977b03eb